packages/substrate-client/src/chainhead/follow.ts

```diff
@@ -40,36 +44,51 @@ export function follow(
       | ((event: FollowEventWithoutRuntime) => void)
       | ((event: FollowEventWithRuntime) => void),
   ): FollowResponse => {
+    const subscriptions = getSubscriptionsManager()
     const _genesisHash = new Promise<string>((res) => {
       request("chainHead_unstable_genesisHash", [], res)
     })
-
     let unfollow: () => void = () => {}
     let followSubscription: Promise<string> | string = new Promise((res) => {
       unfollow = request(
         "chainHead_unstable_follow",
         [withRuntime],
-        (result: string, follow) => {
-          follow(
-            (event, done) => {
-              if (event.event === "stop") done()
-              cb(event as any)
+        (subscriptionId, follow) => {
+          const done = follow(subscriptionId, {
+            next: (event) => {
+              if ((event as any).operationId) {
+                subscriptions.next((event as any).operationId, event)
```

Comment:
i think a guard or a switch statement could be used here instead of coercing to `as any`. Can just leave a `TODO` here to fix this up.

Reply:
good point! There is a slightly better way to go though 😉 

```ts
function isOpeartionEvent(
  event: FollowEventWithRuntime | FollowEventWithoutRuntime,
): event is OperationEvents {
  return (event as OperationEvents).operationId !== undefined
}
```

and then 🌈 :

```ts
if (isOpeartionEvent(event)) {
  subscriptions.next(event.operationId, event)
}
```